app/_src/guides/gateway-delegated.md

```diff
@@ -0,0 +1,197 @@
+
+In the [quickstart](/docs/{{ page.version }}/quickstart/kubernetes-demo/), traffic was only able to get in the mesh by port-forwarding to an instance of an app
+inside the mesh.
+In production, you typically set up a gateway to receive traffic external to the mesh.
+In this guide you will add Kong as a [delegated gateway](/docs/{{ page.version }}/using-mesh/managing-ingress-traffic/delegated/) in front of the demo-app service and expose it publicly.
+
+{% mermaid %}
+---
+title: service graph of the demo app with a Kong gateway on front
+---
+flowchart LR
+  subgraph Kong Gateway 
+    gw0(/ :80)
+  end
+  demo-app(demo-app :5000)
+  redis(redis :6379)
+  gw0 --> demo-app 
+  demo-app --> redis
+{% endmermaid %}
+
+## Prerequisites
+- Completed [quickstart](/docs/{{ page.version }}/quickstart/kubernetes-demo/) to set up a zone control plane with demo application
+
+## Install Kong ingress controller 
+
+Follow the steps on the [Kong docs website](https://docs.konghq.com/kubernetes-ingress-controller/latest/get-started/) to install the ingress controller.
+
+{% warning %}
+The Kubernetes cluster needs to support LoadBalancer for this to work.
+This may not be the case if the Kubernetes cluster is running locally with `kind` or `k3d`.
+{% endwarning %}
+
+## Enable sidecar injection on the `kong` namespace
+
+The Kong Ingress controller was installed outside the mesh.
+For it to work as a delegated gateway restart it with [sidecar injection enabled](/docs/{{ page.version }}/production/dp-config/dpp-on-kubernetes/):
+
+Add the label:
+```shell
+kubectl label namespace kong kuma.io/sidecar-injection=enabled
+```
+
+Restart both the controller and the gateway to leverage sidecar injection:
+```shell
+kubectl rollout restart -n kong deployment kong-gateway kong-controller
+```
+
+Wait until pods are fully rollout and look at them:
+```shell
+kubectl get pods -n kong
+```
+
+It is now visible that both pods have 2 containers, one for the application and one for the sidecar.
+```shell
+NAME                              READY   STATUS    RESTARTS      AGE
+kong-controller-675d48d48-vqllj   2/2     Running   2 (69s ago)   72s
+kong-gateway-674c44c5c4-cvsr8     2/2     Running   0             72s
+```
+
+Verify the gateway still works:
+```shell
+curl -i $PROXY_IP
+```
+
+outputs that there are no routes defined:
+```shell
+HTTP/1.1 404 Not Found
+Date: Fri, 09 Feb 2024 15:25:45 GMT
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+Content-Length: 103
+X-Kong-Response-Latency: 0
+Server: kong/3.5.0
+X-Kong-Request-Id: e7dfe659c9e46639a382f82c16d9582f
+
+{
+  "message":"no Route matched with those values",
+  "request_id":"e7dfe659c9e46639a382f82c16d9582f"
+}%
+```
+
+## Add a route to our `demo-app`
+
+Patch our gateway to allow routes in any namespace:
+
+```shell
+kubectl patch --type=json gateways.gateway.networking.k8s.io kong -p='[{"op":"replace","path": "/spec/listeners/0/allowedRoutes/namespaces/from","value":"All"}]'
+```
```

Comment:
Mostly for personal understanding (but might also be worth adding something), I wondered:

1. If we had to do this just because we are using kong i.e. is this necessary "because kong" or "because kuma"? Guessing its "because kong"?
2. Pretty sure this is allowing kong `HTTPRoute`s in any namespace, might be worth this is allowing kong routes not kuma ones.

I think as I'm writing this I'm kinda answering my own questions in my head and things are making more sense. I think generally it might be worth adding and underlining the things that are different specifically because this is a delegated 3rd party gateway, umm I suppose the "why we are doing these bits".

Reply:
It's actually GatewayAPI

Reply:
Oooh I see interesting!